File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        unordered_map<int, int> degree, from;
        for (auto con : trust) {
            degree[con[1]]++;
            from[con[0]]++;
        }
        int ans = -1;
        for (int i = 1; i <= n; i++) {
            if (degree[i] == (n-1) && from[i] == 0) {
                if (ans != -1) return -1;
                ans = i;
            }
        }
        return ans;
    }
};

File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] inDegree = new int[n + 1];
        int[] outDegree = new int[n + 1];
        for (int i = 0; i < trust.length; i++) {
            inDegree[trust[i][1]]++;
            outDegree[trust[i][0]]++;
        }
        int ans = -1, cnt = 0;
        for (int i = 1; i <= n; i++) {
            if (inDegree[i] == (n-1) && outDegree[i] == 0) {
                ans = i;
                cnt++;
            }
        }
        if (cnt > 1) return -1;
        return ans;
    }
}